### PR TITLE
Added new formatter for week of the month

### PR DESCRIPTION
**Pull request by minitesh**
Added a new formatter to get week of the month.
With respect to issue [#1134](https://github.com/date-fns/date-fns/issues/1134)

### CODE REVIEW

**Comment by kossnocorp** [comment]
There's a weird issue in tests that makes them fail. I'm going to investigate it.

**Comment by minitesh** [comment]
@kossnocorp 
I thought it is because I am using `getWeekOfMonth` function directly instead of writing a separate function to get UTC month of the week.
Is that so?

**Comment by kossnocorp** [comment]
Yup, that's the problem. I'm currently working on adding `getUTCWeekOfMonth`. If you have time to continue now, I'll gladly give you the chance because I'm just started 😌

**Comment by minitesh** [comment]
No problem.
I can work on it tomorrow if you are ok with it.

**Comment by kossnocorp** [comment]
All righty, I'll leave it to you. Thank you! I'll need to add the UTC version of `getWeekOfMonth` and all functions that used in there (`getDate`, `getDay` and `startOfMonth`). These functions will be part of the public API in the future, so tests are also necessary.

**Comment by minitesh** [comment]
Ok
Thanks @kossnocorp

**Comment by minitesh** [comment]
@kossnocorp 
I have added the required UTC versions of all the methods and tests are now passing.
Meanwhile I will start working on tests for the same.

**Comment by kossnocorp** [comment]
@minitesh good work, keep it up!

File: src/_lib/getUTCDateOfMonth/index.js [hunk 1]
```diff
@@ -2,7 +2,7 @@ import toDate from '../../toDate/index.js'
 
 // This function will be a part of public API when UTC function will be implemented.
 // See issue: https://github.com/date-fns/date-fns/issues/376
-export default function getUTCDate(dirtyDate) {
+export default function getUTCDateOfMonth(dirtyDate) {
```
Comment (leshakoss) [inline src/_lib/getUTCDateOfMonth/index.js]: Hi! Please rename `getUTCDateOfMonth` back to `getUTCDate`. [`getDate`](https://date-fns.org/v2.0.0-beta.1/docs/getDate) is the name of date-fns's function, we took the name from JS' standard library: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Date/getDate

File: src/_lib/getUTCDayOfWeek/index.js [hunk 2]
```diff
@@ -2,14 +2,14 @@ import toDate from '../../toDate/index.js'
 
 // This function will be a part of public API when UTC function will be implemented.
 // See issue: https://github.com/date-fns/date-fns/issues/376
-export default function getUTCDay(dirtyDate) {
+export default function getUTCDayOfWeek(dirtyDate) {
```
Comment (leshakoss) [inline src/_lib/getUTCDayOfWeek/index.js]: Same for `getUTCDay`: https://date-fns.org/v2.0.0-beta.1/docs/getDay https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Date/getDay

File: src/_lib/getUTCDateOfMonth/index.js (hunk 1 above)
Comment (minitesh) [reply src/_lib/getUTCDateOfMonth/index.js]: Ok
Done

File: src/_lib/getUTCDayOfWeek/index.js (hunk 2 above)
Comment (minitesh) [reply src/_lib/getUTCDayOfWeek/index.js]: Done

**Comment by minitesh** [comment]
@leshakoss and @kossnocorp
Just a reminder

**Comment by RobertShaw1** [comment]
@leshakoss is there anything left to do on this?

**Comment by minitesh** [comment]
@leshakoss and @kossnocorp
Can you please review this again?

**Pull request closed without merging**